diff --git a/aliyun/src/test/java/org/apache/iceberg/aliyun/oss/mock/AliyunOSSMock.java b/aliyun/src/test/java/org/apache/iceberg/aliyun/oss/mock/AliyunOSSMock.java
index 7894c1857..f5aa5906a 100644
--- a/aliyun/src/test/java/org/apache/iceberg/aliyun/oss/mock/AliyunOSSMock.java
+++ b/aliyun/src/test/java/org/apache/iceberg/aliyun/oss/mock/AliyunOSSMock.java
@@ -80,31 +80,45 @@ public class AliyunOSSMock {
       String[] requests = request.split("/");
       String bucketName = requests[0];
       if (requests.length == 1) {
-        // bucket operations
-        if (httpExchange.getRequestMethod().equals("PUT")) {
-          putBucket(bucketName, httpExchange);
-        }
-        if (httpExchange.getRequestMethod().equals("DELETE")) {
-          deleteBucket(bucketName, httpExchange);
-        }
+        handleBucketOperation(bucketName, httpExchange);
       } else {
-        // object operations
         String objectName = requests[1];
         if (objectName.contains("?")) {
           objectName = objectName.substring(0, objectName.indexOf("?"));
         }
-        if (httpExchange.getRequestMethod().equals("PUT")) {
+        handleObjectOperation(bucketName, objectName, httpExchange);
+      }
+    }
+
+    private void handleBucketOperation(String bucketName, HttpExchange httpExchange)
+        throws IOException {
+      String method = httpExchange.getRequestMethod();
+      switch (method) {
+        case "PUT":
+          putBucket(bucketName, httpExchange);
+          break;
+        case "DELETE":
+          deleteBucket(bucketName, httpExchange);
+          break;
+      }
+    }
+
+    private void handleObjectOperation(String bucketName, String objectName, HttpExchange httpExchange)
+        throws IOException {
+      String method = httpExchange.getRequestMethod();
+      switch (method) {
+        case "PUT":
           putObject(bucketName, objectName, httpExchange);
-        }
-        if (httpExchange.getRequestMethod().equals("DELETE")) {
+          break;
+        case "DELETE":
           deleteObject(bucketName, objectName, httpExchange);
-        }
-        if (httpExchange.getRequestMethod().equals("HEAD")) {
+          break;
+        case "HEAD":
           getObjectMeta(bucketName, objectName, httpExchange);
-        }
-        if (httpExchange.getRequestMethod().equals("GET")) {
+          break;
+        case "GET":
           getObject(bucketName, objectName, httpExchange);
-        }
+          break;
       }
     }
 
@@ -129,6 +143,7 @@ public class AliyunOSSMock {
             createErrorResponse(
                 OSSErrorCode.BUCKET_NOT_EMPTY, "The bucket you tried to delete is not empty.");
         handleResponse(httpExchange, 409, errorMessage, "application/xml");
+        return;
       }
       handleResponse(httpExchange, 200, "OK", "application/xml");
     }
@@ -203,75 +218,90 @@ public class AliyunOSSMock {
 
       Object range = httpExchange.getRequestHeaders().get("Range");
       if (range != null) {
-        range = range.toString().replace("[bytes=", "").replace("]", "");
-        String[] ranges = range.toString().split("-");
-        long rangeStart = -1;
-        if (!ranges[0].isEmpty()) {
-          rangeStart = Long.parseLong(ranges[0]);
-        }
-        long rangeEnd = -1;
-        if (ranges.length == 2 && !ranges[1].isEmpty()) {
-          rangeEnd = Long.parseLong(ranges[1]);
-        }
-        if (rangeEnd == -1) {
-          rangeEnd = Long.MAX_VALUE;
-          if (rangeStart == -1) {
-            rangeStart = 0;
-          }
-        }
+        handleRangeRequest(bucketName, filename, metadata, range, httpExchange);
+      } else {
+        handleFullRequest(metadata, httpExchange);
+      }
+    }
 
-        long fileSize = metadata.getContentLength();
-        long bytesToRead = Math.min(fileSize - 1, rangeEnd) - rangeStart + 1;
-        long skipSize = rangeStart;
+    private void handleRangeRequest(
+        String bucketName,
+        String filename,
+        ObjectMetadata metadata,
+        Object range,
+        HttpExchange httpExchange)
+        throws IOException {
+      range = range.toString().replace("[bytes=", "").replace("]", "");
+      String[] ranges = range.toString().split("-");
+      long rangeStart = -1;
+      if (!ranges[0].isEmpty()) {
+        rangeStart = Long.parseLong(ranges[0]);
+      }
+      long rangeEnd = -1;
+      if (ranges.length == 2 && !ranges[1].isEmpty()) {
+        rangeEnd = Long.parseLong(ranges[1]);
+      }
+      if (rangeEnd == -1) {
+        rangeEnd = Long.MAX_VALUE;
         if (rangeStart == -1) {
-          bytesToRead = Math.min(fileSize - 1, rangeEnd);
-          skipSize = fileSize - rangeEnd;
-        }
-        if (rangeEnd == -1) {
-          bytesToRead = fileSize - rangeStart;
-        }
-        if (bytesToRead < 0 || fileSize < rangeStart) {
-          httpExchange.sendResponseHeaders(416, 1);
-          return;
+          rangeStart = 0;
         }
+      }
 
-        httpExchange.getResponseHeaders().add("Accept-Ranges", "bytes");
-        httpExchange
-            .getResponseHeaders()
-            .add(
-                "Content-Range",
-                "bytes "
-                    + rangeStart
-                    + "-"
-                    + (bytesToRead + rangeStart + 1)
-                    + "/"
-                    + metadata.getContentLength());
-        httpExchange.getResponseHeaders().add("ETag", metadata.getContentMD5());
-        httpExchange
-            .getResponseHeaders()
-            .add("Last-Modified", createDate(metadata.getLastModificationDate()));
-        httpExchange.getResponseHeaders().add("Content-Type", metadata.getContentType());
-        httpExchange.getResponseHeaders().add("Content-Length", Long.toString(bytesToRead));
-        httpExchange.sendResponseHeaders(206, bytesToRead);
-        try (OutputStream outputStream = httpExchange.getResponseBody()) {
-          try (FileInputStream fis = new FileInputStream(metadata.getDataFile())) {
-            fis.skip(skipSize);
-            ByteStreams.copy(new BoundedInputStream(fis, bytesToRead), outputStream);
-          }
+      long fileSize = metadata.getContentLength();
+      long bytesToRead = Math.min(fileSize - 1, rangeEnd) - rangeStart + 1;
+      long skipSize = rangeStart;
+      if (rangeStart == -1) {
+        bytesToRead = Math.min(fileSize - 1, rangeEnd);
+        skipSize = fileSize - rangeEnd;
+      }
+      if (rangeEnd == -1) {
+        bytesToRead = fileSize - rangeStart;
+      }
+      if (bytesToRead < 0 || fileSize < rangeStart) {
+        httpExchange.sendResponseHeaders(416, 1);
+        return;
+      }
+
+      httpExchange.getResponseHeaders().add("Accept-Ranges", "bytes");
+      httpExchange
+          .getResponseHeaders()
+          .add(
+              "Content-Range",
+              "bytes "
+                  + rangeStart
+                  + "-"
+                  + (bytesToRead + rangeStart - 1)
+                  + "/"
+                  + metadata.getContentLength());
+      httpExchange.getResponseHeaders().add("ETag", metadata.getContentMD5());
+      httpExchange
+          .getResponseHeaders()
+          .add("Last-Modified", createDate(metadata.getLastModificationDate()));
+      httpExchange.getResponseHeaders().add("Content-Type", metadata.getContentType());
+      httpExchange.getResponseHeaders().add("Content-Length", Long.toString(bytesToRead));
+      httpExchange.sendResponseHeaders(206, bytesToRead);
+      try (OutputStream outputStream = httpExchange.getResponseBody()) {
+        try (FileInputStream fis = new FileInputStream(metadata.getDataFile())) {
+          fis.skip(skipSize);
+          ByteStreams.copy(new BoundedInputStream(fis, bytesToRead), outputStream);
         }
-      } else {
-        httpExchange.getResponseHeaders().add("Accept-Ranges", "bytes");
-        httpExchange.getResponseHeaders().add("ETag", metadata.getContentMD5());
-        httpExchange
-            .getResponseHeaders()
-            .add("Last-Modified", createDate(metadata.getLastModificationDate()));
-        httpExchange.getResponseHeaders().add("Content-Type", metadata.getContentType());
-        httpExchange.sendResponseHeaders(200, metadata.getContentLength());
+      }
+    }
 
-        try (OutputStream outputStream = httpExchange.getResponseBody()) {
-          try (FileInputStream fis = new FileInputStream(metadata.getDataFile())) {
-            ByteStreams.copy(fis, outputStream);
-          }
+    private void handleFullRequest(ObjectMetadata metadata, HttpExchange httpExchange)
+        throws IOException {
+      httpExchange.getResponseHeaders().add("Accept-Ranges", "bytes");
+      httpExchange.getResponseHeaders().add("ETag", metadata.getContentMD5());
+      httpExchange
+          .getResponseHeaders()
+          .add("Last-Modified", createDate(metadata.getLastModificationDate()));
+      httpExchange.getResponseHeaders().add("Content-Type", metadata.getContentType());
+      httpExchange.sendResponseHeaders(200, metadata.getContentLength());
+
+      try (OutputStream outputStream = httpExchange.getResponseBody()) {
+        try (FileInputStream fis = new FileInputStream(metadata.getDataFile())) {
+          ByteStreams.copy(fis, outputStream);
         }
       }
     }
@@ -336,7 +366,7 @@ public class AliyunOSSMock {
    *
    * <p>This code is borrowed from `org.apache.commons:commons-io`
    */
-  public class BoundedInputStream extends FilterInputStream {
+  public static class BoundedInputStream extends FilterInputStream {
 
     /** The max count of bytes to read. */
     private final long maxCount;
